### PR TITLE
Make groups panel collapsible

### LINKED ISSUE #689: Option for hiding the groups sidebar
I'm not using groups and empty sidebar takes up screenspace.
Please add ability to hide sidebar.
Thanks!

### PR DESCRIPTION
**Pull request by rominf**
## Description
Make groups panel collapsible.

## Motivation and context
Fixes #689 

## How has this been tested?
Manually. 

## Types of changes
- ✅ New feature (non-breaking change which adds functionality)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**

### CODE REVIEW

**Comment by phoerious** [comment]
Please do not delete the pull request template.

**Comment by rominf** [comment]
@phoerious Is it OK now?

File: src/gui/DatabaseWidget.cpp [hunk 1]
```diff
@@ -81,12 +81,12 @@ DatabaseWidget::DatabaseWidget(Database* db, QWidget* parent)
     mainLayout->addWidget(m_messageWidget);
     mainLayout->addLayout(layout);
     m_mainSplitter = new QSplitter(m_mainWidget);
-    m_mainSplitter->setChildrenCollapsible(false);
     m_detailSplitter = new QSplitter(m_mainWidget);
     m_detailSplitter->setOrientation(Qt::Vertical);
     m_detailSplitter->setChildrenCollapsible(true);
 
     QWidget* rightHandSideWidget = new QWidget(m_mainSplitter);
+    m_mainSplitter->setCollapsible(0, false);
```
Comment (phoerious) [inline src/gui/DatabaseWidget.cpp]: I would prefer you do it the other way round: leave childrenCollapsible at false and set collapsible to true for child No. 1.

However, I'm generally not a fan of collapsible panels. They tend to disappear with hardly any to no indication where they went. I think this needs some visual indicator telling the user how to get their panel back. What do you think @droidmonkey @TheZ3ro ?

Comment (droidmonkey) [reply src/gui/DatabaseWidget.cpp]: Should be same as the details view.

Comment (phoerious) [reply src/gui/DatabaseWidget.cpp]: Which is simply collapsible, too, and I don't like it

Comment (TheZ3ro) [reply src/gui/DatabaseWidget.cpp]: I think we should have some visual indicator, but so this won't be in 2.3.0

Comment (rominf) [reply src/gui/DatabaseWidget.cpp]: > I would prefer you do it the other way round: leave childrenCollapsible at false and set collapsible to true for child No. 1.
Default value for childrenCollapsible is true. I used this way because of simplicity.

Comment (phoerious) [reply src/gui/DatabaseWidget.cpp]: The default was explicitly set to false and you changed it back to true. I would prefer false as the default. Right now we only have two child widgets, but that may not always be the case.

File: src/core/Entry.cpp [hunk 2]
```diff
@@ -1031,7 +1031,7 @@ QString Entry::resolveUrl(const QString& url) const
 
     // Validate the URL
     QUrl tempUrl = QUrl(newUrl);
-    if (tempUrl.isValid() && (tempUrl.scheme() == "http" || tempUrl.scheme() == "https")) {
+    if (tempUrl.isValid()) {
```
Comment (droidmonkey) [inline src/core/Entry.cpp]: This is totally unrelated change, please revert.

**Comment by droidmonkey** [comment]
Closing this because it is not ready and haven't seen any traction.
